CoThings/Views/Settings/ServerSettingsView.swift

```diff
@@ -27,26 +30,65 @@ struct ServerSettingsView: View {
     }
     
 	var body: some View {
-        Form {
-            Section(header: Text("Server URL")) {
-                HStack(alignment: .firstTextBaseline, spacing: 1) {
-                    Text("https://")
-                    TextField("demo-eu.cothings.app", text: $serverHostname)
-                        .keyboardType(.URL)
-                        .disableAutocorrection(true)
-                        .autocapitalization(.none)
-                        .frame(maxWidth: .infinity)
-                }
+        ZStack{
+            NavigationLink("Scan QR Code", destination: QRCodeView(
+            onFoundCode: { (code) in
+                self.isScanningQRCode = false
+                self.parseQRCode(code: code)
+            },
+            onCameraError: {
+                self.showCameraErrorAlert = true
+            }
+            ), isActive: $isScanningQRCode)
+                .hidden()
+                .alert(isPresented: $showCameraErrorAlert) { () -> Alert in
+                    Alert(title: Text("Error"), message: Text("Cannot open camera.\nMake sure to allow CoThings to access your camera."), dismissButton: .default(Text("Back"), action: {
+                        self.isScanningQRCode = false
+                    }))
             }
-            
-            Section {
-                Button("Done", action: self.save)
+            Form {
+                Section(header: Text("Server URL")) {
+                    GeometryReader { metrics in
```

Comment:
No need for a geometry reader, please use `layoutPriority`'s for text field and label.